flink-runtime/src/main/java/org/apache/flink/runtime/asyncprocessing/AsyncExecutionController.java

```diff
@@ -300,14 +300,29 @@ boolean tryOccupyKey(RecordContext<K> recordContext) {
     @Override
     public <IN, OUT> InternalStateFuture<OUT> handleRequest(
             @Nullable State state, StateRequestType type, @Nullable IN payload) {
+        return handleRequest(state, type, payload, false);
+    }
+
+    /**
+     * Submit a {@link StateRequest} to this AsyncExecutionController and trigger it if needed.
+     *
+     * @param state the state to request. Could be {@code null} if the type is {@link
+     *     StateRequestType#SYNC_POINT}.
+     * @param type the type of this request.
+     * @param payload the payload input for this request.
+     * @param overdraft whether to allow overdraft.
```

Comment:
nit: it would be better to call this param allowOverdraught in line with the was it is referred to elsewhere.

I am not sure what you mean by an overdraught.